tools/extract_features.cpp

```diff
@@ -51,7 +50,7 @@ int feature_extraction_pipeline(int argc, char** argv) {
   arg_pos = num_required_args;
   if (argc > arg_pos && strcmp(argv[arg_pos], "GPU") == 0) {
     LOG(ERROR)<< "Using GPU";
-    uint device_id = 0;
+    unsigned int device_id = 0;
```

Comment:
This probably should have just been `int` all along, yes? `atoi` returns `int`, the check below is currently bogus, and `SetDevice` expects `int`. And we generally don't use `unsigned` per Google style guide.

This is the only appearance of `uint` in Caffe; I wonder if it's a typo!